src/main/java/com/getcloudcherry/survey/SurveyActivity.java

```diff
@@ -257,63 +246,293 @@ void filterQuestions(SurveyResponse iResponse) {
+     *
+     * @param isLastPage boolean to check if view pager is on last page
+     * @param iAnswer    array list of answer object as per API documentation
+     */
+    public void submitAnswerPartial(final boolean isLastPage, ArrayList<Answer> iAnswer) {
+        final ProgressDialog aProgressDialog = new ProgressDialog(SurveyActivity.this);
+        aProgressDialog.setTitle("Submitting survey");
+        aProgressDialog.setMessage("Please wait...");
+        aProgressDialog.setCancelable(false);
+        if (isLastPage)
+            aProgressDialog.show();
+        Call<ResponseBody> aCall = SurveyClient.get().postAnswerPartial(SurveyCC.getInstance().getPartialResponseId(), isLastPage ? true : false, iAnswer);
+        aCall.enqueue(new Callback<ResponseBody>() {
+            @Override
+            public void onResponse(Call<ResponseBody> call, Response<ResponseBody> response) {
+                try {
+                    aProgressDialog.dismiss();
+                    if (isLastPage)
+                        finishSurvey();
+                } catch (Exception e) {
+                    e.printStackTrace();
+                }
+            }
+
+            @Override
+            public void onFailure(Call<ResponseBody> call, Throwable t) {
+                try {
+                    aProgressDialog.dismiss();
+                    if (aAlertDialog != null) {
+                        aAlertDialog.cancel();
                     }
+                    if (isLastPage)
+                        showAlertRetryCallback(RETRY_SUBMIT, getString(R.string.alert_message_survey_submit_failed), SurveyActivity.this);
+                } catch (Exception e) {
+                    e.printStackTrace();
                 }
-            });
+            }
+        });
+    }
+
+    /**
+     * Moves viewpager page to the next item or finishes the survey
+     */
+    public void moveOrSubmit() {
+        if (getCurrentFragment() instanceof MultiPageFragment) {
+            nextPage();
         } else {
-            finishSurvey(aAlert);
+            finishSurvey();
         }
     }
 
-    void finishSurvey(AlertDialog aAlert) {
-        if (aAlert != null) {
-            aAlert.cancel();
+    /**
+     * Method to reset all saved data and show thank you screen
+     */
+    void finishSurvey() {
+        if (aAlertDialog != null) {
+            aAlertDialog.cancel();
         }
         RecordAnswer.getInstance().reset();
         replaceFragment(new ThanksFragment());
     }
 
-    public void showAlertRetryCallback(String iMessage, Context iContext, DialogInterface.OnClickListener iYesListener) {
+    /**
+     * Handles display of next page or submitting answer based on certain criteria
+     */
+    void nextPage() {
+        boolean aIsLastPage = mIsLastPage;
+        if (!aIsLastPage) {
+            getMultiPageViewPager().setCurrentItem(mCurrentPage + 1);
+        } else {
+            if (!SurveyCC.getInstance().isPartialCapturing()) {
```

Comment:
Whether it's partial capturing or not we should submit answers once completed. Partial response submission is just a temporary queue storage and won't reflect immediately. Two API calls should be made while submitting an answer.
#1 Partial Response API Call with complete as true.
#2 Post all answers API Call.